### PR TITLE
Public-Sans: Update Ruby and Node versions

### LINKED ISSUE #307: Public-Sans: Feature: Update ruby & node version
### Is your feature request related to a problem? Please describe.

Update `.ruby-version` and `.nvmrc` to uses latest LTS version.

### Describe the solution you'd like

A newer version of Ruby and Node are used for improved features and security.

### Describe alternatives you've considered

n/a

### Additional context

This should be a regular part of POAM process.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://designsystem.digital.gov/about/community/#community-conduct).
- [X] I checked the [current issues](https://github.com/uswds/public-sans/issues) for duplicate feature requests.

### PR DESCRIPTION
**Pull request by mahoneycm**
# Summary

Updated defined Node and Ruby versions of the repo.

## Related issue

Closes #307 

##  Preview link

[Public-sans →](https://federalist-514d1714-e827-44f8-a29f-3ec1837bf619.sites.pages.cloud.gov/preview/uswds/public-sans/cm-update-node-and-ruby/)

## Problem statement

Newer versions of Ruby and Node are used for improved features and security.

## Solution

1. Update `.nvrmc` to latest LTS version - 20.15.1
2. Update `.ruby-version` to the latest version that's in normal maintenance.
3. Run `gem install bundler` to confirm relevant version of the bundler is installed.

## Testing and review

1. Update your local node version
    1. `nvm install 20.15.1`
    2. `nvm use 20`
2. Update your local ruby version
    1. `rvm install 3.3.4 -- --with-openssl-dir=$(brew --prefix openssl@1.1)`
    2. `rvm use 3.3.4`
1. Run `npm install` and confirm there are no errors
4. Run `bundle install` and confirm there are no errors 
5. Run `npm run build start` and confirm there are no errors
6. Run `npm run serve` and confirm site builds as expected
7. Look at tool versions (listed below) and confirm version decisions

### Tool release schedules

[Node releases →](https://nodejs.org/en/about/previous-releases#nodejs-releases)
[Ruby maintenance branches ->](https://www.ruby-lang.org/en/downloads/branches/)

### CODE REVIEW

File: .nvmrc [hunk 1]
```diff
@@ -1 +1 @@
-18
+20
```
Comment (mejiaj) [inline .nvmrc:1]: **suggestion:** Let's set this to the LTS version so we can always stay up-to-date.

```suggestion
lts/*
```
https://github.com/nvm-sh/nvm?tab=readme-ov-file#long-term-support

**Review by mejiaj — CHANGES_REQUESTED** [review]
Thanks for updating this! Minor suggestion, otherwise looks good. No issues following test commends (the `npm run build` task runs pretty long).

I see that the target branch is `cm-POAM-july-2024` was that intentional? Just want to make sure.

**Comment by mahoneycm** [comment]
@mejiaj sounds good! Committed your suggestion 👍 

And yes, I was hoping to include this with the POAM updates so I set it as the target branch. Happy to change if need be but this was branched off of the POAM updates so the node_modules would be up to date

**Review by mejiaj — APPROVED** [review]
Thanks Charlie.

**Pull request merged**